### PR TITLE
ImageControl component for InspectorControls

### PR DESCRIPTION
**Pull request by shramee**
## Description
ImageControl component for InspectorControls

## How Has This Been Tested?
Tested adding and changing images

## Screenshots (jpeg or gifs if applicable):
image
image

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style.
- [ ] My code has proper inline documentation.

### CODE REVIEW

File: components/image-control/style.scss [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.blocks-textarea-control__input {
+	width: 100%;
+	padding: 6px 8px;
+}
```
Comment (paulwilde) [inline components/image-control/style.scss]: This appears to be styling which applies to the `TextareaControl` rather than this component.

**Comment by aduth** [comment]
Should this be restricted to images, or other types of media generally? (With `type` being configurable as a prop)

File: components/image-control/style.scss (hunk 1 above)
Comment (shramee) [reply components/image-control/style.scss]: Oops, Will fix 🛠

**Comment by shramee** [comment]
Well, yeah, I thought of the same thing and called it `MediaUploadControl` but we show image preview in the control which is not compatible with other `type`s.

I may do controls for video and media gallery later :wink:

File: components/image-control/style.scss (hunk 1 above)
Comment (shramee) [reply components/image-control/style.scss]: Done ;) Removed the style.scss file...

Comment (shramee) [reply components/image-control/style.scss]: @paulwilde Fixed the error in code...

**Comment by danielbachhuber** [comment]
Closing this for now, as it doesn't offer any immediate benefit to Gutenberg itself. Feel free to open a new PR if this abstraction benefits existing controls, etc.

**Pull request closed without merging**